numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -1980,16 +1980,10 @@ array_zeros(PyObject *NPY_UNUSED(ignored), PyObject *args, PyObject *kwds)
 static PyObject *
 array_count_nonzero(PyObject *NPY_UNUSED(self), PyObject *args, PyObject *kwds)
 {
-    PyObject *array_in;
     PyArrayObject *array;
     npy_intp count;
 
-    if (!PyArg_ParseTuple(args, "O", &array_in)) {
-        return NULL;
-    }
-
-    array = (PyArrayObject *)PyArray_FromAny(array_in, NULL, 0, 0, 0, NULL);
-    if (array == NULL) {
+    if (!PyArg_ParseTuple(args, "O&", PyArray_Converter, &array)) {
```

Comment:
Yeah. I'm beginning to see that. I will keep trying though. Even if I succeed eventually, your solution should probably accepted since it provides the correct functionality. A C drop-in replacement implementation should not change the API you propose.